### PR TITLE
[INFRA] Improve documentation of GitHub Actions workflows

### LINKED ISSUE #3543: Document all GitHub Actions workflows
The [README.md](https://github.com/nilearn/nilearn/blob/main/.github/workflows/README.md) in `.github/workflows` only documents the documentation build workflow. It can be useful to comprehensively document all the GiHub Actions workflows here. 

Linking comments: https://github.com/nilearn/nilearn/pull/3536#pullrequestreview-1315426922 and https://github.com/nilearn/nilearn/pull/3536#issuecomment-1446297374

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #3543.

Hopefully this makes it a bit easier to understand our CI infrastructure.

### CODE REVIEW

**Comment by ymzayek** [comment]
I left out `assign.yml` workflow. I haven't seen this used. Can we remove?

**Comment by Remi-Gau** [comment]
> I left out `assign.yml` workflow. I haven't seen this used. Can we remove?

the workflow works

I suspect it was designed to that people without triage access to repo can self assign an issue, is that correct?

**Review by Remi-Gau — APPROVED** [review]

**Comment by ymzayek** [comment]
> I suspect it was designed to that people without triage access to repo can self assign an issue, is that correct?

Ah didn't think about that détail. I suppose but in practice it's never used. What usually happens is that someone just says they'll work on something and one of the maintainers assigns it if anything. 

That being said maybe if it's documented people will use it.

**Comment by Remi-Gau** [comment]
> That being said maybe if it's documented people will use it.

maybe... 

you and others have more experience on the history of the repo than I do to see how much it was used, so I will defer to y'all on this one.

File: .github/workflows/README.md
Comment (RaphaelMeudec) [inline .github/workflows/README.md:68]: Unclear to me what this BROKEN is?

**Review by RaphaelMeudec — COMMENTED** [review]
With some CI general knowledge (and no knowledge of the nilearn one), this documentation feels clear. The only element I don't get is the BROKEN in the circleci artifacts.

File: .github/workflows/README.md
Comment (ymzayek) [reply .github/workflows/README.md:68]: I put that there because it doesn't work. But I agree it's a bit weird. I'll instead leave it out until we get it fixed.

**Review by bthirion — APPROVED** [review]
LGTM.

**Pull request merged**
